File: 0062-unique-paths/0062-unique-paths.go
func uniquePaths(m int, n int) int {
    
    result := make([][]int, m)
    for i:=0; i<m; i++{
        result[i] = make([]int,n)
    }
    
    for i:=0; i<m;i++{
        result[i][0] = 1
    }
    
    for j:=1; j<n; j++{
        result[0][j] = 1
    }
    
    
    for i:=1;i<m; i++{
        for j:=1; j<n; j++{
            result[i][j] = result[i-1][j] + result[i][j-1]
        }
    }
    return result[m-1][n-1]
    
}

File: 0062-unique-paths/0062-unique-paths.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {
        vector<vector<int>> dp(m, vector<int>(n, 1)); // initialize all cells to 1

        for (int i = 1; i < m; i++) {
            for (int j = 1; j < n; j++) {
                // Each cell can be reached either from the cell above it or the cell on its left.
                dp[i][j] = dp[i - 1][j] + dp[i][j - 1];
            }
        }

        // The bottom right cell contains the number of unique paths to reach it.
        return dp[m - 1][n - 1];
    }
};
